LayoutTests/accessibility/aria-label-valuetext-date-input.html

```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
+<html>
+<head>
+<script src="../resources/accessibility-helper.js"></script>
+<script src="../resources/js-test.js"></script>
+</head>
+<body>
+
+<input id="date-input" type="date" value="2022-02-02" />
+
+<script>
+    var testOutput = "Tests that aria-label and aria-valuetext update correctly for date inputs.\n\n";
```

Comment:
I don't think we actually test the aria-label yet 🤔 you should be able to get that with `platformValueForW3CName` (defined in `accessibility-helpers.js`).

What do you think about naming the test something like `date-input-aria-label-and-value.html`? Having date-input up-front reads a bit better IMO, but feel free to tweak that.

Really thorough set of testcases overall, nice job.